### PR TITLE
fix build error and use dep to manage dependencies

### PR DESCRIPTION
**Pull request by bobotu**
In this PR:
* fix build error due to pb change.
* use `dep` to manage dependencies.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+[prune]
+  go-tests = true
+  unused-packages = true
+  non-go = true
+
+  [[prune.project]]
+    name = "github.com/pingcap/tidb"
+    non-go = false
+    unused-packages = false
```
[the remark this answers is not in the document]
Comment (coocood) [reply Gopkg.toml]: The added files maybe much less if we prune unsued packages

[the remark this answers is not in the document]
Comment (coocood) [reply Gopkg.toml]: 913,730 lines is too much.

[the remark this answers is not in the document]
Comment (bobotu) [reply Gopkg.toml]: If so, we still need build script for `parser.go` after vendor update. So I prefer to copy related script from TiDB into Makefile.

[the remark this answers is not in the document]
Comment (coocood) [reply Gopkg.toml]: We can run the build script in `tidb`'s repo rather than `vendor`.

**Comment by bobotu** [comment]
The `parser.go` file is manually generated by:
* checkout TiDB repo to specified revision defined in `Gopkg.lock` 
* `make parser`
* copy and add into this repo

**Comment by coocood** [comment]
LGTM

@ngaut PTAL

**Pull request merged**